apps/server/src/stores/runtimeState.ts

```diff
@@ -677,26 +679,35 @@ export function roll(
 export function loadBlock(rundown: Rundown, state = runtimeState) {
   if (state.eventNow === null) {
     // we need a loaded event to have a block
-    state.currentBlock.block = null;
-    state.currentBlock.startedAt = null;
+    state.blockNow = null;
+    state.blockNext = null;
     return;
   }
 
   const currentBlockId = state.eventNow.parent;
 
-  // update time only if the block has changed
-  if (state.currentBlock.block?.id != currentBlockId) {
-    state.currentBlock.startedAt = null;
+  // find next block
+  const blockOrEventIndex = rundown.order.findIndex((id) => id === currentBlockId || id === state.eventNow?.id);
+  if (blockOrEventIndex >= 0) {
+    for (let i = blockOrEventIndex + 1; i < rundown.order.length; i++) {
+      if (isOntimeBlock(rundown.entries[rundown.order[i]])) {
+        state.blockNext = { id: rundown.order[i], startedAt: null };
+
+        break;
+      }
+    }
   }
 
-  // update the block anyway
+  // not inside a block
   if (currentBlockId === null) {
-    state.currentBlock.block = null;
+    state.blockNow = null;
     return;
   }
 
-  const currentBlock = rundown.entries[currentBlockId];
-  state.currentBlock.block = currentBlock as OntimeBlock;
+  //we went into a new block
+  if ((state.blockNow != null && state.blockNow.id != currentBlockId) || state.blockNow == null) {
+    state.blockNow = { id: currentBlockId, startedAt: null };
```

Comment:
question: why are we unable to set the `startedAt` at this point?

Reply:
that is responsibility of other parts of the runtimeState, here we just load the block other places can diside to start

Reply:
Sounds good, can we make a comment here to say that?